### PR TITLE
Key comparator

### PR DESCRIPTION
**Pull request by raczeja**

### CODE REVIEW

File: src/main/java/uk/co/objectivity/test/db/beans/xml/Compare.java [hunk 1]
```diff
@@ -56,12 +56,18 @@
...
     @XmlAttribute
     private boolean minusQueryIndicatorOn = false;
 
     @XmlElement(name = "sql")
     private List<Sql> sqls;
 
+    @XmlElement(name = "beforeSqls")
+    private List<BeforeSqls> beforeSqls;
+
     @XmlElement(name = "assert")
```
Comment (aimiebell) [inline src/main/java/uk/co/objectivity/test/db/beans/xml/Compare.java:71]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Use of identifier that is a keyword in later versions of Java (NM_FUTURE_KEYWORD_USED_AS_IDENTIFIER)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-use-of-identifier-that-is-a-keyword-in-later-versions-of-java-nm-future-keyword-used-as-identifier).
The identifier is a word that is reserved as a keyword in later versions of Java, and your code will need to be changed in order to compile it in later versions of Java.

File: src/main/java/uk/co/objectivity/test/db/comparators/FileComparator.java [hunk 2]
```diff
@@ -99,19 +99,18 @@ public TestResults getTestResults(TestParams testParams, Connection conn, File f
         File diffFileName = getNewFileBasedOnTestConfigFile(testParams.getTestConfigFile(), "_diff.csv");
         try {
             Sql sql = compare.getSqls().get(0);
-            stmt = conn.prepareStatement(sql.getSql(), ResultSet.TYPE_SCROLL_INSENSITIVE,
-                    ResultSet.CONCUR_READ_ONLY);
-            savedTimes1.StartMeasure("Query "+ sql.getDatasourceName());
+            stmt = conn.prepareStatement(sql.getSql(), ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);
+            savedTimes1.StartMeasure("Query " + sql.getDatasourceName());
```
Comment (aimiebell) [inline src/main/java/uk/co/objectivity/test/db/comparators/FileComparator.java:103]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

Comment (raczeja) [reply src/main/java/uk/co/objectivity/test/db/comparators/FileComparator.java]: fixed, thank you

File: src/main/java/uk/co/objectivity/test/db/beans/xml/Compare.java (hunk 1 above)
Comment (raczeja) [reply src/main/java/uk/co/objectivity/test/db/beans/xml/Compare.java:71]: fixed, thank you

**Pull request merged**